xlators/features/locks/src/posix.c

```diff
@@ -207,24 +216,24 @@ dict_delete_domain_key(dict_t *dict, char *key, data_t *value, void *data)
     return 0;
 }
 
-void
-pl_get_xdata_requests(pl_local_t *local, dict_t *xdata)
+static void
+pl_get_xdata_requests(pl_local_t *local, dict_t *xdata, unsigned char bitfield)
 {
     if (!local || !xdata)
         return;
 
     GF_ASSERT(local->xdata == NULL);
     local->xdata = dict_copy_with_ref(xdata, NULL);
 
-    if (dict_get_sizen(xdata, GLUSTERFS_ENTRYLK_COUNT)) {
+    if (IS_BIT_SET(bitfield, 0)) {
         local->entrylk_count_req = 1;
```

Comment:
Why we don't simply store the `bitfield` and we get rid of these booleans ?

Reply:
Can you please elaborate.

Reply:
Instead of keeping a boolean in `local` for each value we need, we can store the bitfield itself in a single field and then check the corresponding bit instead of the current boolean field.